src/lib/ListMetrics.js

```diff
@@ -19,26 +20,29 @@ export default class ListMetrics {
     /**
      * Create the metric object to retrieve data from schema methods
      * @param {string} resource - The resource to get metrics for
+     * @param {string} component - the component to get a schema object for
+     * (e.g., 's3')
      * @return {object} obj - Object with a key-value pair for a schema method
      */
-    _getSchemaObject(resource) {
-        let type;
-        if (this.metric === 'buckets') {
-            type = 'bucket';
-        } else if (this.metric === 'accounts') {
-            type = 'accountId';
-        }
+    _getSchemaObject(resource, component) {
         const obj = {};
-        obj[type] = resource;
+        const schemaKeys = {
+            buckets: 'bucket',
+            accounts: 'accountId',
+        };
+        obj[schemaKeys[this.metric]] = resource;
         obj.level = this.metric;
+        // Include service to generate key for metric
+        obj.service = component;
```

Comment:
In the S3 config, this property is called `component`, I suppose internally to Utapi it should be just be `service`.